### PR TITLE
conda-build 3, remove clangdev runtime dependency

### PR DESCRIPTION
**Pull request by minrk**
finish up #4

- removes runtime dependency on clangdev
- adds `run_constrained` to get compatibility constraints for users with conda >=4.4
- pinning in conda_build_config.yaml
- toolchain template variables

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,29 +14,29 @@ source:
...
+    - {{ compiler('cxx') }}
+    - cmake
     # "compiling .pyc files" fails without this
     - python
     - clangdev =={{ version }}
     # For downloading libcxxabi
     - curl
+  host: []
 
-  run:
```
Comment (isuruf) [inline recipe/meta.yaml]: Add `llvm-meta == {{ version }}` which does the same thing as `run_constrained`, but works for `conda<4.4`

Comment (minrk) [reply recipe/meta.yaml]: 👍 as run or build?

Comment (isuruf) [reply recipe/meta.yaml]: run. You can remove run_constrained

File: recipe/meta.yaml [hunk 2]
```diff
@@ -14,30 +14,30 @@ source:
...
+  # without pulling them in as dependencies
   run:
-    - clangdev =={{ version }}
+    - llvm-meta =={{ version }}
 
 test:
   requires:
-    - python {{ environ['PY_VER'] + '*' }}  # [win]
+    - clangdev
+    - python  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml]: This line can be removed

Comment (isuruf) [reply recipe/meta.yaml]: Might need a rerender afterwards

**Pull request closed without merging**